### PR TITLE
Loopback model description in angular2 model class

### PR DESCRIPTION
**Pull request by jledun**
My prior idea to get a default value ( #199 )for each model class property was not sufficient because I also need to have the type of the fields.

So I'd like to replace my last pull request #212 by this one which provides much more usefull informations on the models and it's completly based on loopback's models (so less hazardous and  easier to test than PR #212 ! ).

The commit is tested with "npm test", no new test added.

The new function getModelDefinition() is a static function.

This commit was started from a fresh master pull, the modif concerned by PR #212 doesn't exist in that commit, it is a completly new feature.

Please let me know if this suits your requirements.

[EDIT]
Added a Models service, providing a static object listing all custom models.
Then user can call ModelsApi.models[ modelName ].getModelDefinition() for all models.

**From git log:**
```
    add getModelDefinition() in generated model class

    static function providing an object which represents basic properties of
    loopback's models :
    {
      name: 'modelName',
        properties: {
          prop1: {
            name: 'prop1',
            type: '',        // stringified js typee
            default: null    // loopback's model default value (if exists)
          }, {...}
        },
        relations: {
          rel1: {
            name: 'rel1',
            type: ''         // stringified relation type
        }, {...}
      }
    }

    Tested OK
```

**What's new ?**
* lib/angular2/index.js : added buildPropertyDefaultValue() and provide this function to ejs for the models schemas, add Models.ts in schema
* lib/angular2/shared/services/custom/models.ejs : template for Models service
* lib/angular2/shared/models/model.ts : add template for static function getModelDefinition()

Cheers !

### CODE REVIEW

File: tests/angular2/src/app/shared/sdk/services/custom/Models.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { UserCredential } from '../../models/UserCredential';
+import { UserIdentity } from '../../models/UserIdentity';
+import { Category } from '../../models/Category';
+import { Account } from '../../models/Account';
+import { RoomAccount } from '../../models/RoomAccount';
+import { Storage } from '../../models/Storage';
+import { Core } from '../../models/Core';
+
+@Injectable()
+export class ModelsApi {
```
Comment (jonathan-casarrubias) [inline tests/angular2/src/app/shared/sdk/services/custom/Models.ts]: I like a lot the idea of providing the models through a single singleton service, actually it fits quite well with the refactor I'm doing in my side where now I'm providing the real time and storage classes as injectable services.

**Comment by jonathan-casarrubias** [comment]
Hey @jledun, If you want me to add you to the list of contributors, please share with me the following info:

1.- Name
2.- Email
3.- Website

Cheers
jon

**Comment by jledun** [comment]
Hey @jonathan-casarrubias , glad what I've done is ok for you !

Name : Julien Ledun
email : j.ledun@iosystems.fr
website : https://github.com/jledun
http://www.iosystems.fr

Regards,
Julien

**Comment by jonathan-casarrubias** [comment]
Yes good job.

Since I'm going to publish a major release for both FireLoop and the SDK Builder, this enhancements may take some days before they are landed into npm. 

stay tuned.

Cheers
Jon

**Pull request merged**